### PR TITLE
Feature/app 1698 test e2e plugin upgradeability

### PR DESCRIPTION
**Pull request by novaknole**
## Description

Please include a summary of the change and be sure you follow the contributions rules we do provide [here](./CONTRIBUTIONS.md)

Task: [ID]()

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran all tests with success and extended them if necessary.
- [ ] I have updated the ``CHANGELOG.md`` file in the root folder.
- [ ] I have tested my code on the test network.

### CODE REVIEW

File: packages/contracts/test/framework/dao/dao-factory.ts [hunk 1]
```diff
@@ -496,4 +519,196 @@ describe('DAOFactory: ', function () {
...
+        [ownerAddress]
+      );
+
+      let adminPluginInstallation = createPrepareInstallationParams(
+        adminPluginRepoPointer,
+        data
+      );
+      tx = await daoFactory.createDao(daoSettings, [adminPluginInstallation]);
+      event = await findEvent(tx, EVENTS.InstallationPrepared);
+      {
```
Comment (Rekard0) [inline packages/contracts/test/framework/dao/dao-factory.ts:580]: do u need these brackets ?

File: packages/contracts/test/framework/dao/dao-factory.ts [hunk 2]
```diff
@@ -496,4 +519,196 @@ describe('DAOFactory: ', function () {
...
+
+      // Prepare actions for applyUpdate to succeed.
+      let applyUpdateActions = [
+        {
+          to: dao.address,
+          value: 0,
+          data: DAO_INTERFACE.encodeFunctionData('grant', [
+            plugin,
+            psp.address,
+            ethers.utils.id('UPGRADE_PLUGIN_PERMISSION'),
```
Comment (Rekard0) [inline packages/contracts/test/framework/dao/dao-factory.ts:660]: do we also need to revoke this permission at the end of list of actions? and also check it?

Comment (novaknole) [reply packages/contracts/test/framework/dao/dao-factory.ts:660]: No need. We are just testing if plugin can be installed uninstalled updated. Anything else is irrelevant.

**Review by Rekard0 — APPROVED** [review]
LGTM!

File: packages/contracts/test/framework/dao/dao-factory.ts (hunk 1 above)
Comment (novaknole) [reply packages/contracts/test/framework/dao/dao-factory.ts:580]: Yeah, brackets help with code readability. Separates different things from each other. Better than comments

File: packages/contracts/test/framework/dao/dao-factory.ts [hunk 3]
```diff
@@ -496,4 +519,196 @@ describe('DAOFactory: ', function () {
...
+      // Prepare actions for apply Uninstallation.
+      let applyUninstallationActions = [
+        {
+          to: psp.address,
+          value: 0,
+          data: PSP_INTERFACE.encodeFunctionData('applyUninstallation', [
+            dao.address,
+            createApplyUninstallationParams(
+              plugin,
+              [pluginSetupMockRepoAddress, 1, 2],
```
Comment (brickpop) [inline packages/contracts/test/framework/dao/dao-factory.ts:702]: Why is the version number required to uninstall a plugin? Isn't the plugin address and the pluginRepo's address (to resolve the setup) enough?

**Review by brickpop — APPROVED** [review]

**Pull request merged**